File: android/app/src/main/java/com/emops/app/ui/screens/weekly/ScorecardSection.kt
package com.emops.app.ui.screens.weekly

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.emops.app.domain.model.WeeklySheet
import com.emops.app.ui.theme.*

@Composable
fun ScorecardSection(sheet: WeeklySheet?) {
    Column(verticalArrangement = Arrangement.spacedBy(16.dp)) {
        Text("Friday Scorecard", fontWeight = FontWeight.Bold, color = EMOps_Text, fontSize = 18.sp)

        // DORA Metrics
        ScorecardGroup("DORA Metrics", listOf(
            "Deploy Frequency" to (sheet?.scorecard?.dora?.deployFreq),
            "Lead Time" to (sheet?.scorecard?.dora?.leadTime),
            "Change Fail Rate" to (sheet?.scorecard?.dora?.changeFailRate),
            "Time to Restore" to (sheet?.scorecard?.dora?.timeToRestore)
        ))

        // SLO
        ScorecardGroup("SLO Compliance", listOf(
            "Compliance" to (sheet?.scorecard?.slo?.compliance),
            "Error Budget Burn" to (sheet?.scorecard?.slo?.errorBudgetBurn)
        ))

        // SPACE
        ScorecardGroup("SPACE-lite", listOf(
            "Deep Work Hours" to (sheet?.scorecard?.space?.deepWorkHours),
            "Friction Pulse" to (sheet?.scorecard?.space?.frictionPulse)
        ))

        // AI Health
        ScorecardGroup("AI Health", listOf(
            "Assisted %" to (sheet?.scorecard?.aiHealth?.assistedPct),
            "Risk Catches" to (sheet?.scorecard?.aiHealth?.riskCatches)
        ))

        // AI Summary
        if (sheet?.aiWeeklySummary != null) {
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = EMOps_SurfaceVariant)
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text("AI Weekly Summary", color = EMOps_Primary,
                        fontWeight = FontWeight.Bold, fontSize = 16.sp)
                    Spacer(modifier = Modifier.height(8.dp))
                    Text(sheet.aiWeeklySummary, color = EMOps_Text, fontSize = 14.sp)
                }
            }
        }
    }
}

@Composable
private fun ScorecardGroup(
    title: String,
    metrics: List<Pair<String, com.emops.app.domain.model.MetricEntry?>>
) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(containerColor = EMOps_Surface)
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(title, color = EMOps_Primary, fontWeight = FontWeight.SemiBold, fontSize = 14.sp)
            Spacer(modifier = Modifier.height(8.dp))
            metrics.forEach { (name, entry) ->
                Row(
                    modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp),
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Text(name, color = EMOps_Text, fontSize = 13.sp,
                        modifier = Modifier.weight(1f))
                    Text(entry?.thisWeek ?: "-", color = EMOps_TextSecondary, fontSize = 13.sp)
                    Text(" / ${entry?.target ?: "-"}", color = EMOps_TextSecondary, fontSize = 13.sp)
                }
            }
        }
    }
}

File: ios/EMOps/Views/WeeklySheet/ScorecardSection.swift
import SwiftUI

struct ScorecardSection: View {
    let sheet: WeeklySheet

    // DORA
    @State private var deployFreq = EditableMetric()
    @State private var leadTime = EditableMetric()
    @State private var changeFailRate = EditableMetric()
    @State private var timeToRestore = EditableMetric()

    // SLO
    @State private var compliance = EditableMetric()
    @State private var errorBudgetBurn = EditableMetric()

    // SPACE
    @State private var deepWorkHours = EditableMetric()
    @State private var frictionPulse = EditableMetric()

    // AI Health
    @State private var assistedPct = EditableMetric()
    @State private var riskCatches = EditableMetric()

    var body: some View {
        VStack(spacing: 16) {
            // DORA Metrics
            metricGroup(
                title: "DORA Metrics",
                icon: "gauge.with.dots.needle.33percent",
                color: Color(hex: 0x6C8CFF),
                metrics: [
                    ("Deploy Frequency", $deployFreq),
                    ("Lead Time", $leadTime),
                    ("Change Fail Rate", $changeFailRate),
                    ("Time to Restore", $timeToRestore)
                ]
            )

            // SLO Metrics
            metricGroup(
                title: "SLO Metrics",
                icon: "chart.bar.xaxis",
                color: Color(hex: 0x00D4AA),
                metrics: [
                    ("Compliance", $compliance),
                    ("Error Budget Burn", $errorBudgetBurn)
                ]
            )

            // SPACE Metrics
            metricGroup(
                title: "SPACE Metrics",
                icon: "person.3",
                color: Color(hex: 0xFFB84D),
                metrics: [
                    ("Deep Work Hours", $deepWorkHours),
                    ("Friction Pulse", $frictionPulse)
                ]
            )

            // AI Health Metrics
            metricGroup(
                title: "AI Health",
                icon: "cpu",
                color: Color(hex: 0x6C8CFF),
                metrics: [
                    ("Assisted %", $assistedPct),
                    ("Risk Catches", $riskCatches)
                ]
            )
        }
        .onAppear {
            guard let scorecard = sheet.scorecard else { return }

            deployFreq = EditableMetric(from: scorecard.dora.deployFreq)
            leadTime = EditableMetric(from: scorecard.dora.leadTime)
            changeFailRate = EditableMetric(from: scorecard.dora.changeFailRate)
            timeToRestore = EditableMetric(from: scorecard.dora.timeToRestore)

            compliance = EditableMetric(from: scorecard.slo.compliance)
            errorBudgetBurn = EditableMetric(from: scorecard.slo.errorBudgetBurn)

            deepWorkHours = EditableMetric(from: scorecard.space.deepWorkHours)
            frictionPulse = EditableMetric(from: scorecard.space.frictionPulse)

            assistedPct = EditableMetric(from: scorecard.aiHealth.assistedPct)
            riskCatches = EditableMetric(from: scorecard.aiHealth.riskCatches)
        }
    }

    private func metricGroup(
        title: String,
        icon: String,
        color: Color,
        metrics: [(String, Binding<EditableMetric>)]
    ) -> some View {
        EMOpsCard {
            VStack(alignment: .leading, spacing: 12) {
                HStack(spacing: 8) {
                    Image(systemName: icon)
                        .foregroundColor(color)
                    Text(title)
                        .font(.headline)
                        .foregroundColor(Color(hex: 0xE8ECF4))
                }

                ForEach(Array(metrics.enumerated()), id: \.offset) { index, item in
                    let (label, binding) = item
                    metricEntryView(label: label, color: color, metric: binding)

                    if index < metrics.count - 1 {
                        Divider()
                            .background(Color(hex: 0x242836))
                    }
                }
            }
        }
    }

    private func metricEntryView(label: String, color: Color, metric: Binding<EditableMetric>) -> some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(label)
                .font(.subheadline)
                .fontWeight(.medium)
                .foregroundColor(color)

            if !metric.wrappedValue.definition.isEmpty {
                Text(metric.wrappedValue.definition)
                    .font(.caption2)
                    .foregroundColor(Color(hex: 0x8B95A8))
                    .italic()
            }

            HStack(spacing: 8) {
                VStack(alignment: .leading, spacing: 2) {
                    Text("THIS WEEK")
                        .font(.caption2)
                        .foregroundColor(Color(hex: 0x8B95A8))
                    TextField("Value", text: metric.thisWeek)
                        .font(.subheadline)
                        .foregroundColor(Color(hex: 0xE8ECF4))
                        .padding(8)
                        .background(Color(hex: 0x242836))
                        .cornerRadius(6)
                }

                VStack(alignment: .leading, spacing: 2) {
                    Text("TARGET")
                        .font(.caption2)
                        .foregroundColor(Color(hex: 0x8B95A8))
                    TextField("Target", text: metric.target)
                        .font(.subheadline)
                        .foregroundColor(Color(hex: 0xE8ECF4))
                        .padding(8)
                        .background(Color(hex: 0x242836))
                        .cornerRadius(6)
                }
            }

            TextField("Notes", text: metric.notes)
                .font(.caption)
                .foregroundColor(Color(hex: 0x8B95A8))
                .padding(8)
                .background(Color(hex: 0x242836))
                .cornerRadius(6)
        }
    }
}

// MARK: - Editable Metric

struct EditableMetric {
    var definition: String = ""
    var thisWeek: String = ""
    var target: String = ""
    var notes: String = ""

    init() {}

    init(from entry: MetricEntry) {
        definition = entry.definition
        thisWeek = entry.thisWeek
        target = entry.target
        notes = entry.notes
    }
}

#Preview {
    ScrollView {
        ScorecardSection(sheet: PreviewData.sampleSheet)
            .padding()
    }
    .background(Color(hex: 0x0F1117))
}
